src/main/kubeconfig-manager.ts

```diff
@@ -11,26 +11,13 @@ export class KubeconfigManager {
   protected configDir = app.getPath("temp");
   protected tempFile: string;
 
-  private constructor(protected cluster: Cluster, protected contextHandler: ContextHandler, protected port: number) { }
+  constructor(protected cluster: Cluster, protected contextHandler: ContextHandler, protected port: number) { }
 
-  static async create(cluster: Cluster, contextHandler: ContextHandler, port: number) {
-    const kcm = new KubeconfigManager(cluster, contextHandler, port);
-
-    await kcm.init();
-
-    return kcm;
-  }
-
-  protected async init() {
-    try {
-      await this.contextHandler.ensurePort();
-      this.tempFile = await this.createProxyKubeconfig();
-    } catch (err) {
-      logger.error(`Failed to created temp config for auth-proxy`, { err });
+  async getPath() {
+    if (!this.tempFile) {
+      await this.init();
```

Comment:
This is the actual fix, rest of this PR is mostly refactoring because static `create` is not needed anymore.